### PR TITLE
[DDCE-2053][SD] Update error scenario handling if javascript is disabled

### PR DESCRIPTION
**Pull request by sarthak83**
# DDCE-2053
Update error messages to a show a generic 4xx page in case javascript is disabled on the browser

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/controllers/UpscanController.scala [hunk 1]
```diff
@@ -36,15 +36,32 @@ import scala.concurrent.{ExecutionContext, Future}
...
-    Future.successful(getGlobalErrorPage)
+    logger.error(s"Upscan Failure. errorCode: $errorCode, errorMessage: $errorMessage, errorRequestId: $errorRequestId")
+    errorCode match {
+      case "InvalidArgument" | "EntityTooLarge" | "EntityTooSmall" =>
+        Future.successful(getFileUploadProblemPage)
+      case _ => Future.successful(getGlobalErrorPage)
+    }
+  }
+
+  def getFileUploadProblemPage()(implicit request: Request[AnyRef], messages: Messages): Result = {
```
Comment (LetMeTestThat) [inline app/controllers/UpscanController.scala:61]: Ideally we would have a unit test to check this method.

File: app/controllers/UpscanController.scala [hunk 2]
```diff
@@ -36,15 +36,32 @@ import scala.concurrent.{ExecutionContext, Future}
...
+                                 override val global_error: views.html.global_error,
+                                 fileUploadProblemView: views.html.file_upload_problem
                                 )(implicit executionContext: ExecutionContext, ersUtil: ERSUtil,
                                   actorSystem: ActorSystem, override val appConfig: ApplicationConfig)
   extends FrontendController(mcc) with Retryable with I18nSupport with BaseController {
 
   val logger: Logger = Logger(getClass)
+
+
   def failure(): Action[AnyContent] = authAction.async { implicit request =>
```
Comment (LetMeTestThat) [inline app/controllers/UpscanController.scala:48]: In terms of unit testing again, I think we should cover:
1. Valid request which matches the first case
2. Valid request which matches the second case
3. Invalid request

Comment (sarthak83) [reply app/controllers/UpscanController.scala:48]: Done!

File: test/controllers/UpscanControllerSpec.scala [hunk 3]
```diff
@@ -71,6 +73,15 @@ class UpscanControllerSpec extends WordSpecLike with Matchers with OptionValues
       val result: Future[Result] = upscanController().failure().apply(fakeRequest)
       status(result) shouldBe Status.INTERNAL_SERVER_ERROR
     }
+
+    "return a 4xx when one of the valid errorCodes is returned" in {
```
Comment (LetMeTestThat) [inline test/controllers/UpscanControllerSpec.scala:77]: Can you change 4xx to 400 please?

Comment (sarthak83) [reply test/controllers/UpscanControllerSpec.scala]: Changed :)

**Pull request merged**